### DISCUSSION TITLE
Considering clj-kondo config (partly) as a tooling-agnostic source of truth

### DISCUSSION THREAD

**Discussion by vemv** [clj-kondo/clj-kondo#1557] [Ideas]
## Context

https://github.com/clojure-emacs/refactor-nrepl/issues/359#issuecomment-1019639865 (and also stuff beyond that particular use case)

## Problem statement

Different tools might need to be configured essentially the same data.

There isn't a standard format for expressing that, in a DRY manner.

## Proposal

Consider `.clj-kondo/config` a "just data" .edn map (as it is already 😄). The nuance being that it would be considered a stable API not to be broken, so that other tools can depend on it.

It would be fine by me to consider only a subset of this config a "public API". The stable keys could be reflected on a doc or such.

There might also be a helper (related to https://github.com/clj-kondo/clj-kondo/discussions/1555) that reads config and trims everything not in the "public API" whitelist.

## Alternatives

Coordinate tooling authors for agreeing on a common .edn format and/or inline metadata.

Cheers - V

**Comment by borkdude** [comment]
I'm not planning to make breaking changes to the configuration, although I regret some choices here and there :). The options are documented in https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md and https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md.

**Reply by vemv** [reply]
Seems fine for the time being! Thanks for the confirmation.
